packages/react/src/components/Combobox/useCombobox.tsx

```diff
@@ -0,0 +1,131 @@
+import React, { useEffect, useState } from 'react';
+
+import type { ComboboxItemProps } from './Item/Item';
+import { ComboboxItem } from './Item/Item';
+
+export type UseComboboxProps = {
+  children: React.ReactNode;
+  input: string;
+  multiple: boolean;
+  activeValues: ValueItemType[];
+  filter: (inputValue: string, label: string, value: string) => boolean;
+};
+
+export type ValueItemType = {
+  value: string;
+  label: string;
+};
+
+export default function useCombobox({
+  children,
+  input,
+  multiple,
+  filter,
+}: UseComboboxProps) {
+  const [open, setOpen] = useState(false);
+  const [values, setValues] = useState<Set<ValueItemType>>(new Set([]));
+
+  // Update all values
+  useEffect(() => {
+    const allValues: ValueItemType[] = [];
+    React.Children.forEach(children, (child) => {
```

Comment:
Suggest using `reduce` here to partition the options, this will remove the need for `filter` in `getRestChildren`

Use the example [here](https://ramdajs.com/docs/#partition) for suggested output (Tuple).